internal/testrunner/runners/system/servicedeployer/factory.go

```diff
@@ -51,6 +51,11 @@ func Factory(options FactoryOptions) (ServiceDeployer, error) {
 			}
 			return NewDockerComposeServiceDeployer([]string{dockerComposeYMLPath}, sv)
 		}
+	case "agent":
+		dockerComposeYMLPath := filepath.Join(serviceDeployerPath, "docker-compose.yml")
+		if _, err := os.Stat(dockerComposeYMLPath); err == nil {
```

Comment:
Umm, if err is not nil here, this is going to fail with `unsupported service deployer (name: agent)`, right? This can be a bit counter intuitive. I would say that if this directory exist, it should contain this file.